tss-esapi/tests/integration_tests/attributes_tests/locality_attributes_tests.rs

```diff
@@ -0,0 +1,192 @@
+            .expect("Failed to build locality attributes");
+
+        let tpma_locality: TPMA_LOCALITY = expected_locality_attributes.into();
+        assert_eq!(
+            locality, tpma_locality,
+            "Locality did not convert inte expected TPMA_LOCALITY value"
+        );
+
+        assert_eq!(
+            expected_locality_attributes,
+            tpma_locality.into(),
+            "The locality attributes converted from TPMA_LOCALITY did not match the expected value"
+        );
+    }
+}
+
+#[test]
+fn test_constants() {
+    assert_eq!(
+        LocalityAttributes::LOCALITY_ZERO,
+        LocalityAttributesBuilder::new()
+            .with_locality(0)
+            .build()
+            .expect("Failed to build locality attributes"),
+        "LOCALITY_ZERO constant does not have the correct value"
+    );
+
+    assert_eq!(
+        LocalityAttributes::LOCALITY_ONE,
+        LocalityAttributesBuilder::new()
+            .with_locality(1)
+            .build()
+            .expect("Failed to build locality attributes"),
+        "LOCALITY_ONE constant does not have the correct value"
+    );
+
+    assert_eq!(
+        LocalityAttributes::LOCALITY_TWO,
+        LocalityAttributesBuilder::new()
+            .with_locality(2)
+            .build()
+            .expect("Failed to build locality attributes"),
+        "LOCALITY_TWO constant does not have the correct value"
+    );
+
+    assert_eq!(
+        LocalityAttributes::LOCALITY_THREE,
+        LocalityAttributesBuilder::new()
+            .with_locality(3)
+            .build()
+            .expect("Failed to build locality attributes"),
+        "LOCALITY_THREE constant does not have the correct value"
+    );
+
+    assert_eq!(
+        LocalityAttributes::LOCALITY_FOUR,
+        LocalityAttributesBuilder::new()
+            .with_locality(4)
+            .build()
+            .expect("Failed to build locality attributes"),
+        "LOCALITY_FOUR constant does not have the correct value"
+    );
+}
+
+#[test]
+fn test_builder_valid_non_extended() {
+    let locality_attributes = LocalityAttributesBuilder::new()
+        .with_localities(&[0, 1, 2, 3, 4])
+        .build()
+        .expect("Failed to build locality attributes");
+
+    assert!(
+        locality_attributes.locality_zero(),
+        "Locality ZERO was not properly set"
+    );
+    assert!(
+        locality_attributes.locality_one(),
+        "Locality ONE was not properly set"
+    );
+    assert!(
+        locality_attributes.locality_two(),
+        "Locality TWO was not properly set"
+    );
+    assert!(
+        locality_attributes.locality_three(),
+        "Locality THREE was not properly set"
+    );
+    assert!(
+        locality_attributes.locality_four(),
+        "Locality FOUR was not properly set"
+    );
+    assert!(
+        !locality_attributes.is_extended(),
+        "Locality attributes unexpectedly indicated extended"
+    );
+}
+
+#[test]
+fn test_builder_valid_extended() {
```

Comment:
Could you add a test for invalid "extended"?

Reply:
No, perhaps a poorly chosen name for the test function. Because there are no invalid extended values. This is how it work all values from ```33u8``` to ```u8::MAX``` will make one or more of the bits between 5:7 to be set and that will make ```is_extended``` return true.

Reply:
Ah, yes, I just meant one where `as_extended` returns an error.

Reply:
Aha! of course I will do that right away.

Reply:
fixed